binder/src/androidTest/java/io/grpc/binder/internal/LeakSafeOneWayBinderTest.java

```diff
@@ -29,28 +30,34 @@
 public final class LeakSafeOneWayBinderTest {
 
   private LeakSafeOneWayBinder binder;
+  private FakeHandler handler = new FakeHandler();
 
-  private int transactionsHandled;
-  private int lastCode;
-  private Parcel lastParcel;
+  static class FakeHandler implements TransactionHandler {
+    int transactionsHandled;
+    int lastCode;
+    Parcel lastParcel;
 
-  @Before
-  public void setUp() {
-    binder = new LeakSafeOneWayBinder((code, parcel) -> {
+    @Override
+    public boolean handleTransaction(int code, Parcel parcel) {
       transactionsHandled++;
       lastCode = code;
       lastParcel = parcel;
       return true;
-    });
+    }
+  }
+
+  @Before
+  public void setUp() {
+    binder = new LeakSafeOneWayBinder(handler);
```

Comment:
Nit: if we populate this field inline, we can also make it `final` and highlight that it doesn't change throughout the test.

Reply:
See my comment below about why test cases should be able to change it (go/unit-testing-practices#setUp)